### DISCUSSION TITLE
Gratitude for Contextual Lexer

### DISCUSSION THREAD

**Discussion by davidmcnabnz** [lark-parser/lark#1322] [General]
While building/debugging a parser for a major project of 'retro-parsing' an antique language, I've noticed how LARK's framework forks off numerous context-specific lexers, rather than using just one.

This is proving to be a major boost. Thanks so much, folks, for making it part of the architecture. It's saving me a ton of painful efforts of accommodating countless spammy edge cases and creating more edge cases along the way. :)

**Comment by erezsh** [comment]
You're welcome! I'm honestly perplexed that this simple technique isn't common. To my knowledge, Lark is the only parser that does it.
